src/services/comparator.ts

```diff
@@ -0,0 +1,165 @@
+import DestinationService from '../interfaces/DestinationService';
+import { ProcessorTransformationRequest, RouterTransformationRequest, Destination } from '../types';
+import ServiceSelector from '../helpers/serviceSelector';
+import tags from '../v0/util/tags';
+import stats from '../util/stats';
+import logger from '../logger';
+import { CommonUtils } from '../util/common';
+import { PlatformError } from '../v0/util/errorTypes';
+const NS_PER_SEC = 1e9;
+
+export default class ComparatorService {
+  private static getTestThreshold(destination: Destination) {
+    return destination.DestinationDefinition?.Config['cdkV2TestThreshold'] || 0;
+  }
+
+
+  public static async compareDestinationServiceAtProc(
+    primaryServiceName: string,
+    secondaryServiceName: string,
+    input: any,
+  ) {
+    const { events } = input;
+    const { version, destination }: { version: string; destination: string } = input;
+    const { requestMetadata }: { requestMetadata: Object } = input;
+    const { feature }: { feature: string } = input;
+
+    const envThreshold = parseFloat(process.env.COMPARISON_TEST || '0');
+    const destThreshold = this.getTestThreshold(events[0].destination);
+    const compareTestThreshold = envThreshold * destThreshold;
+
+    if (
```

Comment:
We need to invoke primary always and if compare test > 0 then only invoke secondary and compare

Reply:
This whole flow is async I have integrated in controller can you check once